src/wallet.cpp

```diff
@@ -2514,59 +2516,79 @@ void CWallet::GetAllReserveKeys(set<CKeyID>& setAddress) const
     }
 }
 
-
-
-
-std::string CWallet::GetAllGridcoinKeys()
+std::vector<std::pair<CBitcoinAddress, CBitcoinSecret>> CWallet::GetAllPrivateKeys(CWallet* pBackupWallet, std::string& sError)
 {
-    //<CKeyID>& setAddress;
-    //setAddress.clear();
-    std::string backup = "";
     CWalletDB walletdb(strWalletFile);
 
     LOCK2(cs_main, cs_wallet);
-    std::string errors = "";
 
-    BOOST_FOREACH(const int64_t& id, setKeyPool)
+    std::vector<std::pair<CBitcoinAddress, CBitcoinSecret>> res;
+    // Get Privaty Keys from mapAddressBook
+    for (auto const& item : mapAddressBook)
+    {
+        const CBitcoinAddress& address = item.first;
+        bool fMine = ::IsMine(*pBackupWallet, address.Get());
```

Comment:
If someone knows how I don't have to pass the wallet through for this :IsMine() check let me know. this checks to make sure the address is the wallets before trying to pull the private keys